### PR TITLE
HARMONY-1463: Removed collection references from services.yml for all services.

### PR DESCRIPTION
**Pull request by ygliuvt**
## Jira Issue ID
HARMONY-1463, HARMONY-1432

## Description
Removed collection references from services.yml for all services.

General approach:
1. Make the least amount of update on services.yml
2. Assume all existing associations between umm-s records and collections are legit. I am NOT going to remove any existing associations between umm-s records and collections.
3. Add umm-s and collection associations for any collections explicitly listed in the services.yml or implicitly via different umm-s records in services.yml via associations.
4. Use XYZ_PROV umm-s records for DAS services in PROD and EEDTEST umm-s records for DAS services in UAT.
5. For podaac/l2-subsetter service, use POCLOUD umm-s record as the official record in services.yml. Added associations of GES_DISC and LARC_ASDC collections to it. I didn't remove the GES_DISC or LARC_ASDC umm-s records or any collection associations with them. It is up to the GES_DISC and LARC_ASDC providers to do the cleanup. Or I can help with the cleanup with the providers' direction if necessary.

## Local Test Steps
Test with any harmony requests locally with CMR properly configured (UAT or PROD) and verify that the same backend service is picked and the request succeed as before. 

## PR Acceptance Checklist
* [x] Acceptance criteria met
* [x] Tests added/updated (if needed) and passing
* [ ] Documentation updated (if needed)

### CODE REVIEW

File: config/services.yml
Comment (chris-durbin) [inline config/services.yml:394]: Similar comment that this does not look like a UMM-S record for Giovanni:

> The service record corresponding to the Earthdata Cloud OPeNDAP service.  The functionality provided should match or exceed what is provided by our on-prem OPeNDAP servers for these datasets, such as spatial subsetting, variable subsetting, and format conversion.

File: config/services.yml
Comment (chris-durbin) [inline config/services.yml:677]: We might want to update this description for the UMM-S record to indicate it's for testing harmony http service integration (the service isn't docker-based or make GDAL calls - it does not actually do anything).

> Access to Harmony internal Docker-based GDAL service for AMSR-E/Aqua L2B Global Swath Ocean Products.

**Review by chris-durbin — COMMENTED** [review]
As far as harmony is concerned I think this looks fine, but it looks like we used service records meant for other purposes (not harmony). We should work with GESDISC to get the correct service records for Giovanni (or a harmony specific Giovanni service record) and use those instead. If we created any collection associations to the other UMM-S records that are not meant for harmony, we should put them back to what they originally were.

**Review by indiejames — APPROVED** [review]

**Comment by chris-durbin** [comment]
Can you back out the commit (42a0fc3) for the other ticket from this PR? I want to get this one merged in, but there are some things to do for the other ticket that I don't want to hold up for this PR.

**Review by chris-durbin — APPROVED** [review]
I'm good with the changes for HARMONY-1463 - we can merge that in. I have some comments for HARMONY-1432 and HARMONY-1433. We should review those in a separate PR so that we can get these initial changes in.

**Pull request merged**
